apt_offline_core/AptOfflineCoreLib.py

```diff
@@ -937,6 +938,32 @@ def fetcher( args ):
                 opener = urllib.request.build_opener(urllib.request.HTTPSHandler(context=context))
                 urllib.request.install_opener(opener)
                 log.verbose("SSL Client Authentication successfully set up with certificate file %s and key file %s\n" % (Str_HttpsCertFile, Str_HttpsKeyFile))
+
+        if Str_HttpBasicAuth:
+                # create a password manager
+                password_mgr = urllib.request.HTTPPasswordMgrWithDefaultRealm()
+
+                try:
+                        for authstr in Str_HttpBasicAuth:
+                                # Add the username and password.
+                                username = authstr.split("@")[0].split(":")[0]
+                                password = authstr.split("@")[0].split(":")[1]
```

Comment:
THis goes with the assumption that you do not have a username/password without special character `@`